test/test.py

```diff
@@ -7,160 +7,230 @@
+                    self.assertTrue(tensor.allclose(tensor2))
+                    self.assertEqual(sample_rate, sample_rate2)
+                    os.unlink(output_path)
 
     def test_4_load_partial(self):
-        num_frames = 101
-        offset = 201
-        # load entire mono sinewave wav file, load a partial copy and then compare
-        input_sine_path = os.path.join(self.test_dirpath, 'assets', 'sinewave.wav')
-        x_sine_full, sr_sine = torchaudio.load(input_sine_path)
-        x_sine_part, _ = torchaudio.load(input_sine_path, num_frames=num_frames, offset=offset)
-        l1_error = x_sine_full[:, offset:(num_frames + offset)].sub(x_sine_part).abs().sum().item()
-        # test for the correct number of samples and that the correct portion was loaded
-        self.assertEqual(x_sine_part.size(1), num_frames)
-        self.assertEqual(l1_error, 0.)
-        # create a two channel version of this wavefile
-        x_2ch_sine = x_sine_full.repeat(1, 2)
-        out_2ch_sine_path = os.path.join(self.test_dirpath, 'assets', '2ch_sinewave.wav')
-        torchaudio.save(out_2ch_sine_path, x_2ch_sine, sr_sine)
-        x_2ch_sine_load, _ = torchaudio.load(out_2ch_sine_path, num_frames=num_frames, offset=offset)
-        os.unlink(out_2ch_sine_path)
-        l1_error = x_2ch_sine_load.sub(x_2ch_sine[:, offset:(offset + num_frames)]).abs().sum().item()
-        self.assertEqual(l1_error, 0.)
-
-        # test with two channel mp3
-        x_2ch_full, sr_2ch = torchaudio.load(self.test_filepath, normalization=True)
-        x_2ch_part, _ = torchaudio.load(self.test_filepath, normalization=True, num_frames=num_frames, offset=offset)
-        l1_error = x_2ch_full[:, offset:(offset + num_frames)].sub(x_2ch_part).abs().sum().item()
-        self.assertEqual(x_2ch_part.size(1), num_frames)
-        self.assertEqual(l1_error, 0.)
-
-        # check behavior if number of samples would exceed file length
-        offset_ns = 300
-        x_ns, _ = torchaudio.load(input_sine_path, num_frames=100000, offset=offset_ns)
-        self.assertEqual(x_ns.size(1), x_sine_full.size(1) - offset_ns)
-
-        # check when offset is beyond the end of the file
-        with self.assertRaises(RuntimeError):
-            torchaudio.load(input_sine_path, offset=100000)
+        for backend in ["sox"]:
+            with self.subTest():
+                with AudioBackendScope(backend):
+                    num_frames = 101
+                    offset = 201
+                    # load entire mono sinewave wav file, load a partial copy and then compare
+                    input_sine_path = os.path.join(
+                        self.test_dirpath, "assets", "sinewave.wav"
+                    )
+                    x_sine_full, sr_sine = torchaudio.load(input_sine_path)
+                    x_sine_part, _ = torchaudio.load(
+                        input_sine_path, num_frames=num_frames, offset=offset
+                    )
+                    l1_error = (
```

Comment:
I don't want to be "that guy", but these code format changes make this harder to review. You could make them the last commit and continue to maintain that, or you could send a separate PR later on. They also introduce a lot of meaningless git blame changes.

Reply:
Yeah, (1) this was definitely not meant to be in, and (2) this was not quite ready for review :)